fastn.com/podcast/new-fastn-architecture.ftd

```diff
@@ -0,0 +1,277 @@
+"entity") in the decentralized network.
+
+-- ds.h3: Types of Entities
+
+There are three types of entities, each with their own ID52 (public-private key
+pair):
+
+1. **Rig**: The entity you talk to for controlling the fastn run server 2.
+**Accounts**: Represent either a user or organization  3. **Devices**:
+Individual devices belonging to an account
+
+-- ds.h2: Storage Structure
+
+The fastn home folder contains:
+- **rig/**: Contains rig configuration and keys
+- **accounts/**: Account-specific data
+- **public/**: fastn applications and files
+
+Each entity stores its ID52 (public key) and private key, with fallback
+mechanisms for key storage including keychain integration.
+
+-- ds.h2: Accounts vs Devices
+
+Every fastn machine has at least one account. Each account represents either a
+user or organization - we strongly recommend against group accounts where
+multiple people share access.
+
+**Accounts** should have high uptime because:
+- When accounts are down, devices can't sync data with each other
+- Emails sent to you will bounce during downtime
+- Optimal service requires accounts to be online most of the time
+
+**Devices** belong to accounts and can have more relaxed uptime requirements.
+We recommend running accounts on:
+- Cloud servers (Linode, DigitalOcean, AWS)
+- Raspberry Pi or desktop at home/office with reliable internet and power
+- Our hosted fastn cloud service
+
+-- ds.h2: Email Architecture
+
+Every fastn account has email capabilities. When you create an account, fastn
+automatically starts IMAP, POP3, and SMTP protocol servers.
+
+-- ds.h3: Email Organization
+
+Key decisions we've made:
+- Each alias you create is an email domain
+- You can share different email addresses with different people
+- We organize emails by usernames, not by aliases
+- Most users want a single mailbox with multiple aliases rather than multiple
+  separate mailboxes
+
+-- ds.h3: Address Format
+
+Email addresses use the format: `username@alias-id52` or
+`username@alias-id52.com` (for compatibility with email clients expecting .com
+domains).
+
+-- ds.h3: Email Storage
+
+- Emails are stored in account folders, not device folders
+- This prevents email duplication (10GB in fastn folder + 10GB copied by email
+  clients)
+- Devices sync with accounts for email access
+
+-- ds.h2: Automerge Documents
+
+We're automerge first, meaning we use automerge documents as our fundamental
+feature. Anything you store uses key-value pairs where each value is a document
+that can easily sync between instances.
+
+Configuration and metadata are stored in automerge because syncing is
+automatic. For example, if you change your name on your phone, it updates on
+all your devices without custom syncing code.
+
+-- ds.h2: Document Organization
+
+Documents have paths that encode ownership:
+- Your documents use your account ID as prefix
+- Shared documents use the sharing alias as prefix
+- Special documents use `/dash` prefix for system configuration
```

Comment:
/-/ prefix

Reply:
fixed in https://github.com/fastn-stack/fastn/pull/2178/commits/fdde94ba27fcd724ca62a1bd013a78dda8fb378e